### PR TITLE
Add support for VS 2017

### PR DESCRIPTION
**Pull request by wangii**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
+
+CMAKE_MINIMUM_REQUIRED(VERSION 3.6)
```
Comment (bjornblissing) [inline CMakeLists.txt]: Is this change really required?

File: CMakeLists.txt [hunk 2]
```diff
@@ -26,6 +27,8 @@ IF (MSVC)
     SET(COMPILER "v120") # Visual Studio 2013
   ELSEIF (${MSVC_VERSION} EQUAL 1900)
     SET(COMPILER "v140") # Visual Studio 2015
+   ELSEIF (${MSVC_VERSION} EQUAL 1910)
+     SET(COMPILER "v141") # Visual Studio 2017
```
Comment (bjornblissing) [inline CMakeLists.txt]: Shouldn't this be 'v150'? Isn' t 'v141' the compiler version for the 2017 beta?

File: CMakeLists.txt (hunk 1 above)
Comment (wangii) [inline CMakeLists.txt]: Probably not.

File: CMakeLists.txt (hunk 2 above)
Comment (wangii) [inline CMakeLists.txt]: I installed the VS 2017 community edition, not RC. and it's v141. odd.

Comment (bjornblissing) [inline CMakeLists.txt]: Well, according to this blog from Microsoft we both may be correct:
https://blogs.msdn.microsoft.com/vcblog/2016/02/24/stuck-on-an-older-toolset-version-move-to-visual-studio-2015-without-upgrading-your-toolset/

The versions number schema for Visual Studio is highly confusing:

| Visual Studio Product Name |  Product Version | Compiler Version |Toolset Version |
| ------------------- |----:| ----:|----:|
| Visual Studio 2008 | 9.0 | 1500 | V90 |
| Visual Studio 2010 | 10.0 | 1600 | V100 |
| Visual Studio 2012 | 11.0 | 1700 | V110 |
| Visual Studio 2013 | 12.0 | 1800 | V120 |
| Visual Studio 2015 | 14.0 | 1900 | V140 |
| Visual Studio 2017 | 15.0 | 1910 | V141 |

But I guess what I am actually referring to is the toolset version. So your version is probably the most correct, although I should probably rename the variable `COMPILER` to `TOOLSET`.

**Comment by bjornblissing** [comment]
I have created a new issue for this (#41).
